From 7c8f68c5428380b930579dc9ef27c853264448fd Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 15 May 2017 15:06:11 +0300
Subject: [PATCH] Prepend PKG_CONFIG_SYSROOT_DIR to pkg-config output

In cross environment we have to prepend the sysroot to the path found by
pkgconfig since the path returned from pkgconfig does not have sysroot prefixed
it ends up using the files from host system. If build host has wayland installed
the build will succeed but if you dont have wayland-protocols installed on build
host then it wont find the files on build host

This should work ok with non sysrooted builds too since
in those cases PKG_CONFIG_SYSROOT_DIR will be empty

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index e307be6..83cdeb0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2272,7 +2272,7 @@ AG_GST_CHECK_FEATURE(WAYLAND, [wayland sink], wayland , [
   PKG_CHECK_MODULES(WAYLAND, wayland-client >= 1.4.0 libdrm >= 2.4.55 wayland-protocols >= 1.4, [
     if test "x$wayland_scanner" != "x"; then
       HAVE_WAYLAND="yes"
-      AC_SUBST(WAYLAND_PROTOCOLS_DATADIR, `$PKG_CONFIG --variable=pkgdatadir wayland-protocols`)
+      AC_SUBST(WAYLAND_PROTOCOLS_DATADIR, ${WAYLAND_PROTOCOLS_SYSROOT_DIR}`$PKG_CONFIG --variable=pkgdatadir wayland-protocols`)
     else
       AC_MSG_RESULT([wayland-scanner is required to build the wayland plugin])
       HAVE_WAYLAND="no"
-- 
2.4.0

